### PR TITLE
Documentation update 042021

### PR DESCRIPTION
**Pull request by cpfergus1**
This PR updates the  Getting Started and Customization chapters of the edge guides. Few grammatical changes were required and the examples required some syntax changing to get to properly working on a fresh install of rails. Additionally an extra sub chapter has been included to inform users how to setup their local RSpec Testing Suite to  properly follow along with the examples in those chapters.

Another PR will be created for the Advanced chapters when completed. The Optional Installation sub-chapter is also going to contain installation for Redis and Sidekiq, but the subsections have not been created yet!

### CODE REVIEW

File: getting-started/what-is-solidus.md [hunk 1]
```diff
@@ -24,36 +24,32 @@ For maximum flexibility, you can decide you just want to install specific gems a
...
-Then, install the bundle and configure Solidus:
 
 ```bash
-$ bundle install
+$ bundle add 'solidus' --version '~>3.0.0'
 $ bin/rails generate solidus:install
 ```
 
-The installer will prompt you for an admin email and password. You can leave the default \(email: admin@example.com, password: test123\) or enter your own.
+The installer will prompt you on a few questions before completing the installation. First it will ask if you would like to use the default authentication solution, [Devise](https://github.com/heartcombo/devise), or your implement your own. Next it will ask what payment service you would like to install with solidus. Currently, Solidus only supports Paypal and is the default option. You may install your own and avoid implementing Paypal by typing 'none' when prompted. Finally, you will be prompted for an admin email and password. You can leave the default \(email: admin@example.com, password: test123\) or enter your own.
```
Comment (kennyadsl) [inline getting-started/what-is-solidus.md:47]: Instead of suggesting to install "your own payment method", we may want to link to the [extensions](https://solidus.io/extensions/) page, which contains a list of alternative payment methods in the Payment section. WDYT?

File: customization/customizing-the-core.md [hunk 2]
```diff
@@ -170,6 +170,8 @@ end
 {% tab title="order\_finalization\_notifier\_spec.rb" %}
 {% code title="spec/lib/awesome\_store/order\_finalization\_notifier\_spec.rb" %}
 ```ruby
+require "awesome_store/order_finalization_notifier.rb"
```
Comment (kennyadsl) [inline customization/customizing-the-core.md:173]: `amazing`?

File: customization/customizing-the-api.md [hunk 3]
```diff
@@ -128,18 +128,20 @@ Let's create our controller, along with its view, route a request spec to test i
 
 {% tabs %}
 {% tab title="product\_likes\_controller.rb" %}
-{% code title="app/controllers/api/product\_likes\_controller.rb" %}
+{% code title="app/controllers/api/products/product\_likes\_controller.rb" %}
```
Comment (kennyadsl) [inline customization/customizing-the-api.md:131]: Why this change? I think it should match the module inside of it, so `app/controllers/api/product\_likes\_controller` was looking good to me. Any reason to change that I'm missing? Maybe they are both missing the `spree`/`Spree` folder/module?

File: customization/customizing-the-core.md [hunk 4]
```diff
@@ -170,6 +170,8 @@ end
 {% tab title="order\_finalization\_notifier\_spec.rb" %}
 {% code title="spec/lib/awesome\_store/order\_finalization\_notifier\_spec.rb" %}
```
Comment (kennyadsl) [inline customization/customizing-the-core.md:171]: amazing?

File: customization/customizing-the-core.md [hunk 5]
```diff
@@ -259,16 +261,16 @@ Here's our `AddGlobalHiddenFlag` decorator for `Spree::Product`, along with its
 
 {% tabs %}
 {% tab title="add\_global\_hidden\_flag.rb" %}
-{% code title="app/decorators/awesome\_store/spree/product/add\_global\_hidden\_flag.rb" %}
+{% code title="app/decorators/awesome\_store/spree/product/add\_global\_hidden\_flag\_decorator.rb" %}
```
Comment (kennyadsl) [inline customization/customizing-the-core.md:264]: amazing?

File: customization/customizing-the-core.md [hunk 6]
```diff
@@ -259,16 +261,16 @@ Here's our `AddGlobalHiddenFlag` decorator for `Spree::Product`, along with its
...
 ```ruby
 module AwesomeStore
```
Comment (kennyadsl) [inline customization/customizing-the-core.md:266]: `AmazingStore`?

File: getting-started/deployment.md [hunk 7]
```diff
@@ -6,6 +6,10 @@ Deploying a Solidus application to production is no different from deploying any
 
 ### Heroku
 
+{% hint style="warning" %}
+Heroku [does not support](https://devcenter.heroku.com/articles/sqlite3) the rails default database sqlite3. Heroku natively utilizes [PostgreSQL](https://www.postgresql.org/) and adaptively uses other [data storage services](https://elements.heroku.com/addons).
```
Comment (kennyadsl) [inline getting-started/deployment.md:10]: Let's use `Rails` with capital `R`. Also,`Sqlite3` with capital `S`.

I think we should provide a solution along with the problem. For example, we may want to suggest changing the database for production only:

```
# In Gemfile:

group :development, :test do
  gem 'sqlite3'
end

group :production do
  gem 'pg'
end

# In config/database.yml

production:
  adapter: postgresql
  encoding: unicode
  pool: <%= ENV.fetch("RAILS_MAX_THREADS") { 5 } %>
  database: myapp_production
  username: myapp
  password: <%= ENV['MYAPP_DATABASE_PASSWORD'] %>

```

**Review by kennyadsl — COMMENTED** [review]
Left some comments, thanks!

File: getting-started/what-is-solidus.md (hunk 1 above)
Comment (cpfergus1) [reply getting-started/what-is-solidus.md:47]: > Instead of suggesting to install "your own payment method", we may want to link to the [extensions](https://solidus.io/extensions/) page, which contains a list of alternative payment methods in the Payment section. WDYT?

I Agree with this and have updated this line of text.

File: getting-started/deployment.md (hunk 7 above)
Comment (cpfergus1) [reply getting-started/deployment.md]: Updated proper noun capitalization and fixed "Awesome" to "Amazing." Thank you for catching that!

File: customization/customizing-the-api.md (hunk 3 above)
Comment (cpfergus1) [reply customization/customizing-the-api.md]: You are right, the name spacing should be spree/api/product_likes_controller.rb
Updated to reflect the change

File: customization/customizing-the-api.md [hunk 8]
```diff
@@ -268,24 +279,29 @@ Instead, Solidus provides a more manageable way to add attributes to API resourc
...
   before { stub_authentication! }
 
   describe 'GET /:slug' do
     it 'exposes the likes_count field' do
       product = create(:product)
-
       get spree.api_product_path(product)
 
       parsed_response = JSON.parse(response.body)
+      byebug
```
Comment (waiting-for-dev) [inline customization/customizing-the-api.md:304]: I think it sneaked here :slightly_smiling_face:

```suggestion
```

File: getting-started/deployment.md [hunk 9]
```diff
@@ -47,14 +51,52 @@ When deploying a Solidus store, there are a few external dependencies that must
 ### File storage
 
 {% hint style="warning" %}
-Paperclip [has been deprecated](https://github.com/thoughtbot/paperclip#deprecated) in favor of Rails' native [ActiveStorage](https://guides.rubyonrails.org/active_storage_overview.html) library. Solidus already supports ActiveStorage, but due to limitations in ActiveStorage itself you will not be able to use it until Rails 6.1 is released with support for [public URLs](https://edgeguides.rubyonrails.org/active_storage_overview.html#public-access).
+Paperclip [has been deprecated](https://github.com/thoughtbot/paperclip#deprecated) in favor of Rails' native [Active Storage](https://guides.rubyonrails.org/active_storage_overview.html) library. Should you be utilizing a version of rails before 6.1, your application will not be able to use Active Storage and therefore should use Paperclip.
```
Comment (waiting-for-dev) [inline getting-started/deployment.md:54]:
```suggestion
Paperclip [has been deprecated](https://github.com/thoughtbot/paperclip#deprecated) in favor of Rails' native [Active Storage](https://guides.rubyonrails.org/active_storage_overview.html) library. Should you be utilizing a version of Rails before 6.1, your application will not be able to use Active Storage and therefore should use Paperclip.
```

File: getting-started/deployment.md [hunk 10]
```diff
@@ -47,14 +51,52 @@ When deploying a Solidus store, there are a few external dependencies that must
...
+{% endcode %}
+
+Next, it is necessary to edit the production environment Active Storage configuration to use your newly created service. Navigate to your production environment configuration and modify the following variable:
+
+{% code title="config/environment/production.rb" %}
+```ruby
+...
+# Store uploaded files on the local file system (see config/storage.yml for options).
+config.active_storage.service = :amazon
+
```
Comment (waiting-for-dev) [inline getting-started/deployment.md:86]:
```suggestion
```

File: getting-started/deployment.md [hunk 11]
```diff
@@ -47,14 +51,52 @@ When deploying a Solidus store, there are a few external dependencies that must
...
+```
+{% endcode %}
+
+Finally, put your S3 credentials in the environment variables used in `storage.yml`.
+
+If you're not using S3, Active Storage provides support for the most popular storage services, either natively or through third-party plugins.
+
+That is it, you have just set up Active Storage!
+
+### Paper Clip
```
Comment (waiting-for-dev) [inline getting-started/deployment.md:97]: Just wondering... do we need to talk about deprecated stuff in the edge guides? cc @kennyadsl

File: getting-started/optional_installation_settings.md [hunk 12]
```diff
@@ -0,0 +1,220 @@
+# Optional Installation Settings
+## Testing Support Setup
+
+Some of our tests we present in the guide examples utilize non-standard requirements that can be appended to your local rails helper. These additions will help provide an error free learning experience! Let's start with our gemfile to make sure we have required dependencies:
```
Comment (waiting-for-dev) [inline getting-started/optional_installation_settings.md:4]:
```suggestion
Some of the tests we present in the guide examples utilize non-standard requirements that can be appended to your local rails helper. These additions will help provide an error-free learning experience! Let's start with our Gemfile to make sure we have the required dependencies:
```

File: getting-started/optional_installation_settings.md [hunk 13]
```diff
@@ -0,0 +1,220 @@
...
+# Adds support for Capybara system testing and selenium driver
+  gem 'capybara', '>= 3.26'
+  # Easy installation and use of web drivers to run system tests with browsers
+  gem 'webdrivers'
+  # Scrubs database clean between tests
+  gem 'database_cleaner-active_record'
+```
+{% endcode %}
+
+Let's go over installation of these gems and why we are suggesting them!
```
Comment (waiting-for-dev) [inline getting-started/optional_installation_settings.md:29]:
```suggestion
Let's go over the installation of these gems and why we are suggesting them!
```

File: getting-started/optional_installation_settings.md [hunk 14]
```diff
@@ -0,0 +1,220 @@
...
+### RSpec
+RSpec is highly versatile testing framework that we utilize for Solidus and in our guide examples. To install it for your local project, simply Update your bundle and `Gemfile.lock` then install RSpec base files with the following commands in your shell:
```
Comment (waiting-for-dev) [inline getting-started/optional_installation_settings.md:31]:
```suggestion
RSpec is a highly versatile testing framework that we utilize for Solidus and in our guide examples. To install it for your local project, simply update your bundle and install RSpec base files with the following commands in your shell:
```

I think it's weird to talk about updating `Gemfile.lock` as it seems to suggest to do it manually. Do you agree?

File: getting-started/optional_installation_settings.md [hunk 15]
```diff
@@ -0,0 +1,220 @@
...
+
+```bash
+bundle install
```
Comment (waiting-for-dev) [inline getting-started/optional_installation_settings.md:34]: Shouldn't we first add `rspec` to the `Gemfile`?

File: getting-started/optional_installation_settings.md [hunk 16]
```diff
@@ -0,0 +1,220 @@
...
+rails g rspec:install
+```
+{% hint style="info" %}
+You may have to navigate to your `spec_helper.rb` located in the `spec` directory of your applications root folder and make sure that you have the following at the top of the file:
+  {% code title="spec/spec\_helper.rb" %}
+  ```ruby
+  ENV["RAILS_ENV"] ||= 'test'
+  require File.expand_path("../../config/environment", __FILE__)
```
Comment (waiting-for-dev) [inline getting-started/optional_installation_settings.md:42]: Shouldn't it be `require_relative`?

File: getting-started/optional_installation_settings.md [hunk 17]
```diff
@@ -0,0 +1,220 @@
...
+  #...
+  ```
+  {% endcode %}
+{% endhint %}
+
+For more information on RSpec and associated methods, please see the [official documentation](https://relishapp.com/rspec/rspec-rails/docs).
+
+### Solidus Factories and authentication helpers
+
+Whether you are following along this learning guide and coding along with the examples, or developing on top of the Solidus platform, you might find yourself interested in utilizing some of the pre-built factories or authentication helpers. In order use the pre-built factories, we will have to edit a few files starting with your Gemfile:
```
Comment (waiting-for-dev) [inline getting-started/optional_installation_settings.md:53]:
```suggestion
Whether you are following along this learning guide and coding along with the examples or developing on top of the Solidus platform, you might find yourself interested in utilizing some of the pre-built factories or authentication helpers. In order use the pre-built factories, we will have to edit a few files starting with your Gemfile:
```

File: getting-started/optional_installation_settings.md [hunk 18]
```diff
@@ -0,0 +1,220 @@
...
+RSpec.configure do |config|
+  #...
+  config.include FactoryBot::Syntax::Methods
+end
+```
+{% endcode %}
+
+### Byebug
+
+[Byebug](https://github.com/deivid-rodriguez/byebug) is debugging tool which will help you investigate and understand code while it executes. There is no other installation instructions required other than to simply insert `byebug` method anywhere in the code base while you run a test or navigate through your development server. This will pause the execution of code where the method was inserted and provide you with a debugging interface in your shell. We highly recommend familiarizing yourself with byebug's [guide](https://github.com/deivid-rodriguez/byebug/blob/master/GUIDE.md) and [commands](https://github.com/deivid-rodriguez/byebug#byebugs-commands) to learn how to get the most out of this powerful tool.
```
Comment (waiting-for-dev) [inline getting-started/optional_installation_settings.md:119]:
```suggestion
[Byebug](https://github.com/deivid-rodriguez/byebug) is debugging tool which will help you investigate and understand code while it executes. There is no installation instructions required other than to simply insert `byebug` method anywhere in the codebase while you run a test or navigate through your development server. This will pause the execution of code where the method was inserted and provide you with a debugging interface in your shell. We highly recommend familiarizing yourself with byebug's [guide](https://github.com/deivid-rodriguez/byebug/blob/master/GUIDE.md) and [commands](https://github.com/deivid-rodriguez/byebug#byebugs-commands) to learn how to get the most out of this powerful tool.
```

File: getting-started/optional_installation_settings.md [hunk 19]
```diff
@@ -0,0 +1,220 @@
...
+### Capybara and Webdrivers
+
+[Capybara](https://github.com/teamcapybara/capybara) helps you test web applications by simulating how a real user would interact with your app. This is complimented by the `webdrivers` gem to provide selenium drivers required by Capybara to test multiple browsers.  However, some setup is required to utilize the features we will be testing.
```
Comment (waiting-for-dev) [inline getting-started/optional_installation_settings.md:122]:
```suggestion
[Capybara](https://github.com/teamcapybara/capybara) helps you test web applications by simulating how a real user would interact with your app. This is complimented by the `webdrivers` gem to provide selenium drivers required by Capybara to test multiple browsers. However, some setup is required to utilize the features we will be testing.
```

File: getting-started/optional_installation_settings.md [hunk 20]
```diff
@@ -0,0 +1,220 @@
...
+
+First, let's setup our support file structure for RSpec. From the application's root folder, navigate to spec and create a new folder `support/helpers` and in that new folder, create a new file `capybara_config.rb` with the following contents:
+
+{% code title="spec/helpers/capybara\_config.rb" %}
```
Comment (waiting-for-dev) [inline getting-started/optional_installation_settings.md:126]: I'm reluctant to add installation instructions for third-party gems, as it adds a burden to us to keep them updated. Thoughts @kennyadsl ?

File: getting-started/optional_installation_settings.md [hunk 21]
```diff
@@ -0,0 +1,220 @@
...
+  Capybara::Selenium::Driver.new app,
+    browser: :chrome,
+    desired_capabilities: capabilities
+end
+
+Capybara.javascript_driver = :headless_chrome
+```
+{ %endcode% }
+
+This configuration will set the default browser to a headless [Chrome](https://www.google.com/chrome/) window when running your tests. To learn about other supported drivers, please refer to capybara's [guide](https://github.com/teamcapybara/capybara/blob/2.12.0/README.md#using-capybara-with-rspec).
```
Comment (waiting-for-dev) [inline getting-started/optional_installation_settings.md:150]:
```suggestion
This configuration will set the default browser to a headless [Chrome](https://www.google.com/chrome/) window when running your tests. To learn about other supported drivers, please refer to Capybara's [guide](https://github.com/teamcapybara/capybara/blob/2.12.0/README.md#using-capybara-with-rspec).
```

File: getting-started/optional_installation_settings.md [hunk 22]
```diff
@@ -0,0 +1,220 @@
...
+
+Finally, we need to make sure that when `rails_helper` is called during our tests, we will be able to pull this support file. To accomplish this we will need to navigate to our rails helper file and uncomment out or add the following:
+
+
```
Comment (waiting-for-dev) [inline getting-started/optional_installation_settings.md:154]:
```suggestion
```

**Review by waiting-for-dev — COMMENTED** [review]
First-class job @cpfergus1 :raised_hands: Just left some suggestions.

File: getting-started/deployment.md (hunk 11 above)
Comment (kennyadsl) [reply getting-started/deployment.md:97]: Paperclip is not the default anymore but it's not yet deprecated so I'd keep it around in the doc for a while.

File: getting-started/optional_installation_settings.md (hunk 20 above)
Comment (kennyadsl) [reply getting-started/optional_installation_settings.md:126]: I agree, if we can link to projects' specific instruction instead it would be better. Unless we have some special configuration to do for Solidus to work.

File: getting-started/optional_installation_settings.md (hunk 16 above)
Comment (cpfergus1) [reply getting-started/optional_installation_settings.md:42]: This is the way it is created when running rails g rspec:install - I removed this note, however, after going back and cleaning everything up. the failure was with my _spec.rb's not containing require 'rails_helper'

File: getting-started/optional_installation_settings.md (hunk 20 above)
Comment (cpfergus1) [reply getting-started/optional_installation_settings.md:126]: I removed most of the installation instructions, but left the built code as an example of the setup I used to get it to work. Let me know if this is still too much and if I should remove it all together. 
Does this comment extend to DatabaseCleaner section as well?

Thanks!

File: getting-started/optional_installation_settings.md (hunk 15 above)
Comment (cpfergus1) [reply getting-started/optional_installation_settings.md:34]: At the beginning of the guide, I list the Gemfile with all the gems that we will be using. Does that feel too out of place?

Comment (waiting-for-dev) [reply getting-started/optional_installation_settings.md]: Sorry, it's ok. It's just that reviewing chunk by chunk makes it difficult to keep the context, but looking at the file, it's clear.

File: getting-started/optional_installation_settings.md (hunk 20 above)
Comment (waiting-for-dev) [reply getting-started/optional_installation_settings.md]: Yes, in my view, it applies to DatabaseCleaner, too. But you're right that if we're following a manual, then we need to add the code.

File: customization/customizing-the-api.md [hunk 23]
```diff
@@ -268,21 +279,25 @@ Instead, Solidus provides a more manageable way to add attributes to API resourc
 {% code title="config/initializers/spree.rb" %}
 ```ruby
 # ...
-Spree::Api::ApiHelpers.product_attributes << :likes_count
+Spree::Api::ApiHelpers.product_attributes << :like_count
```
Comment (kennyadsl) [inline customization/customizing-the-api.md:282]: isn't this `likes_count`?

File: customization/customizing-the-backend.md [hunk 24]
```diff
@@ -65,59 +64,61 @@ Our new business logic is pretty easy to test in integration:
 
 {% code title="spec/models/spree/order\_spec.rb" %}
 ```ruby
+require 'rails_helper'
+require 'amazing_store/order_analyzer'
```
Comment (kennyadsl) [inline customization/customizing-the-backend.md:68]: I'd like to explore with you why we need to require this module manually. It shouldn't be needed.

File: customization/customizing-the-backend.md [hunk 25]
```diff
@@ -65,59 +64,61 @@ Our new business logic is pretty easy to test in integration:
...
 
 ## Implementing new actions
 
-In order to allow admins to remove an order from the blacklist, we'll add a button to the order detail page that will trigger a new controller action.
+In order to allow admins to remove an order from the rejected, we'll add a button to the order detail page that will trigger a new controller action.
 
 The first step is to add our custom action to `Spree::Admin::OrdersController`. We'll use a decorator to accomplish that:
 
-{% code title="app/decorators/awesome\_store/spree/admin/oders\_controller/add\_remove\_from\_blacklist\_action.rb" %}
+{% code title="app/decorators/amazing\_store/spree/admin/orders\_controller/add\_remove\_from\_rejected\_action\_decorator.rb" %}
```
Comment (kennyadsl) [inline customization/customizing-the-backend.md:110]: Why is `_decorator` added here?

File: customization/customizing-the-core.md [hunk 26]
```diff
@@ -259,16 +262,16 @@ Here's our `AddGlobalHiddenFlag` decorator for `Spree::Product`, along with its
 
 {% tabs %}
 {% tab title="add\_global\_hidden\_flag.rb" %}
-{% code title="app/decorators/awesome\_store/spree/product/add\_global\_hidden\_flag.rb" %}
+{% code title="app/decorators/amazing\_store/spree/product/add\_global\_hidden\_flag\_decorator.rb" %}
```
Comment (kennyadsl) [inline customization/customizing-the-core.md:265]: same question here about `_decorator` addition.

File: getting-started/deployment.md [hunk 27]
```diff
@@ -47,14 +51,51 @@ When deploying a Solidus store, there are a few external dependencies that must
...
 
-Solidus supports storage services out of the box by integrating with the [Paperclip](https://github.com/thoughtbot/paperclip) gem. In order to configure Paperclip, just create an initializer like the following:
+Solidus versions earlier than 3.0.0 support integration with the [Paperclip](https://github.com/thoughtbot/paperclip) gem. In order to configure Paperclip, just create an initializer like the following:
```
Comment (kennyadsl) [inline getting-started/deployment.md:98]: Even Solidus > 3 supports Paperclip, but it's no more the default.

File: SUMMARY.md [hunk 28]
```diff
@@ -5,6 +5,7 @@
 ## Getting Started
 
 * [Installing Solidus](getting-started/what-is-solidus.md)
+  *  [Optional Installation Settings](getting-started/optional_installation_settings.md)
```
Comment (kennyadsl) [inline SUMMARY.md:8]: Should we update this menu item as well?

File: getting-started/deployment.md [hunk 29]
```diff
@@ -6,6 +6,10 @@ Deploying a Solidus application to production is no different from deploying any
 
 ### Heroku
 
+{% hint style="warning" %}
+Heroku [does not support](https://devcenter.heroku.com/articles/sqlite3) the Rails default database Sqlite3. Heroku natively utilizes [PostgreSQL](https://www.postgresql.org/) and adaptively uses other [data storage services](https://elements.heroku.com/addons).
```
Comment (kennyadsl) [inline getting-started/deployment.md:10]: I think we should point out how to use Sqlite3 in development and Postgres in production.

File: getting-started/testing-setup.md [hunk 30]
```diff
@@ -0,0 +1,220 @@
...
+RSpec.configure do |config|
+  #...
+  config.include FactoryBot::Syntax::Methods
+end
+```
+{% endcode %}
+{% endtab %}
+{% endtabs %}
+
+## Byebug
```
Comment (kennyadsl) [inline getting-started/testing-setup.md:127]: `byebug` is not related to testing, it is used in development as well. I think it shouldn't be part of this section.

File: getting-started/testing-setup.md [hunk 31]
```diff
@@ -0,0 +1,220 @@
...
+{% code title="spec/support/capybara.rb" %}
+```ruby
+require 'capybara/rspec'
+require 'selenium/webdriver'
+
+Capybara.default_max_wait_time = 5
+
+Capybara.register_driver :chrome do |app|
+  Capybara::Selenium::Driver.new(app, browser: :chrome)
+end
```
Comment (kennyadsl) [inline getting-started/testing-setup.md:145]: I think specs will work even if we remove this line, we are never using this driver.

File: getting-started/testing-setup.md [hunk 32]
```diff
@@ -0,0 +1,220 @@
...
+```ruby
+module AmazingStore
+  class Application < Rails::Application
+
+    # … everything else
+
+    # Don't initialize this if factory_bot_rails is not part of the current bundle group.
+    if defined?(FactoryBotRails)
+      initializer after: "factory_bot.set_factory_paths" do
+        require 'spree/testing_support'
```
Comment (kennyadsl) [inline getting-started/testing-setup.md:58]: I think the file you required is actually [just requiring](https://github.com/solidusio/solidus/blob/25daae38b6c87931c26506a0948b37f47e00156d/core/lib/spree/testing_support.rb#L7-L10) the stuff we have in [`spree/testing_support/factory_bot`](https://github.com/solidusio/solidus/blob/master/core/lib/spree/testing_support/factory_bot.rb).  It may be better to load that file instead, cc'ing @elia for confirmation since he knows this part better. 

Out of curiosity @cpfergus1: where did you take this code? We may want to update that source as well.

**Review by kennyadsl — COMMENTED** [review]
Left some other questions/comments. 🐻

File: getting-started/testing-setup.md (hunk 32 above)
Comment (cpfergus1) [reply getting-started/testing-setup.md:58]: I got it from Elia's discussion thread [#4014](https://github.com/solidusio/solidus/discussions/4014)

File: customization/customizing-the-backend.md (hunk 25 above)
Comment (cpfergus1) [reply customization/customizing-the-backend.md:110]: On a new solidus application install, the installer sets up the configuration to look for decorators using this syntax:

config/application.rb
` 
...
initializer 'spree.decorators' do |app|
    config.to_prepare do
       Dir.glob(Rails.root.join('app/**/*_decorator*.rb')) do |path|
          require_dependency(path)
       end
   end
end
...
`
perhaps it was intended as:
`
initializer 'spree.decorators' do |app|
    config.to_prepare do
       Dir.glob(Rails.root.join('app/decorators/**/*.rb')) do |path|
          require_dependency(path)
       end
   end
end
`
What do you think?

Comment (kennyadsl) [reply customization/customizing-the-backend.md:110]: Yes, I guess we should change that in core. Want to submit a PR for that after trying if that works in your application?

Comment (cpfergus1) [reply customization/customizing-the-backend.md]: submitted  [#4054](https://github.com/solidusio/solidus/pull/4054)

File: customization/customizing-the-backend.md (hunk 24 above)
Comment (kennyadsl) [reply customization/customizing-the-backend.md:68]: This should not be required anymore after loading the whole `/lib` folder in the `eager_load_paths`, isn't it?

**Review by kennyadsl — COMMENTED** [review]
There seem to be a couple of leftovers. Other than that, we should be good to go!

File: getting-started/deployment.md (hunk 10 above)
Comment (cpfergus1) [reply getting-started/deployment.md]: fixed

File: getting-started/optional_installation_settings.md (hunk 12 above)
Comment (cpfergus1) [reply getting-started/optional_installation_settings.md]: fixed

File: customization/customizing-the-api.md (hunk 23 above)
Comment (cpfergus1) [reply customization/customizing-the-api.md]: fixed

File: customization/customizing-the-backend.md (hunk 24 above)
Comment (cpfergus1) [reply customization/customizing-the-backend.md]: Moved files to app/models as discussed

**Review by kennyadsl — APPROVED** [review]
